packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -175,6 +182,42 @@ public void onMapLoaded() {
         });
   }
 
+  /**
+   * Initializes map renderer if meta-data value for key
+   * "io.flutter.plugins.googlemaps.PREFERRED_RENDERER" is set in AndroidManifest. Renderer can be
+   * initialized only once per application context.
+   *
+   * <p>Supported meta-data values are "legacy" and "latest".
+   */
+  private void initializePreferredRenderer(Context context) {
+    if (!preferredRendererInitialized) {
+      try {
+        ApplicationInfo applicationInfo =
+            context
+                .getPackageManager()
+                .getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+
+        if (applicationInfo != null && applicationInfo.metaData != null) {
+          final String preferredRenderer =
+              applicationInfo.metaData.getString(
+                  "io.flutter.plugins.googlemaps.PREFERRED_RENDERER");
+
+          if (preferredRenderer != null) {
+            switch (preferredRenderer) {
+              case "latest":
+                MapsInitializer.initialize(context, Renderer.LATEST, null);
+              case "legacy":
+                MapsInitializer.initialize(context, Renderer.LEGACY, null);
+            }
+          }
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
```

Comment:
Exceptions should be returned to Dart as channel errors, not unconditionally printed and swallowed.

Reply:
This try catch block is unnecessary after moving to the API implementation. Possible errors will be returned to Dart as channel errors.